VueJS/index.js

```diff
@@ -1,13 +1,72 @@
-import List from './components/List.js'
-import ListItem from './components/ListItem.js'
+// import List from './components/List.js'
+// import ListItem from './components/ListItem.js'
 
-new Vue({
-  el: '#app',
-  data: {
-    todos: [
-      { id: '1', message: 'Foo', },
-      { id: '2', message: 'Bar', },
-      { id: '3', message: 'Baz', }
-    ]
+Vue.component('list-item', {
+  props: ['todo'],
+  template: `
+      <div v-if="edit">
+        <li>
+          <input id="input" v-model="newTodo">
+          <button v-on:click="saveTodo">Save</button>
+          <button v-on:click="cancelTodo">Cancel</button>
+          <button v-on:click="$emit('deleteTodo')">Delete</button>
+        </li>
+      </div>
+      <div v-else>
+        <li>
+        {{ todo.message }}
+        <button v-on:click="editTodo">Edit</button>
+        <button v-on:click="$emit('deleteTodo')">Delete</button>
+        </li>
+      </div>
+  `,
+  data: function(){
+    return{
+      edit: false,
```

Comment:
```suggestion
      isEditing: false,
```
I find this slighly more telling. Another alternative would be `editMode`.